be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -259,9 +264,43 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
     }
   }
 
-// TODO: Proper validation
+  // not implemented since the back end does not receive a ResultElection message coming from the front end
   def validateResultElection(rpcMessage: JsonRpcRequest): GraphMessage = {
-    Right(rpcMessage)
+    def validationError(reason: String): PipelineError = super.validationError(reason, "ResultElection", rpcMessage.id)
+    rpcMessage.getParamsMessage match {
+      case Some(_: Message) =>
+        val (resultElection, laoId, senderPk, channel) = extractData[ResultElection](rpcMessage)
```

Comment:
Since not all of them are used 
```suggestion
        val (resultElection, _, _, channel) = extractData[ResultElection](rpcMessage)
```